lib/persistent_enum.rb

```diff
@@ -126,11 +139,13 @@ def cache_constants(model, required_members, name_attr: :name, required_attribut
 
           # Update or create
           required_members.each do |name, attrs|
-            name = name.to_s
-            attr_names = attrs.keys.map(&:to_s)
+            attr_names = attrs.keys
 
-            if attr_names != table_attributes
-              raise ArgumentError.new("Enum member attribute mismatch: expected #{table_attributes.inspect}, received #{attr_names.inspect}")
+            if (extra_attrs = (attr_names - table_attributes)).present?
+              log_warning("Enum class error: specified attributes #{extra_attrs.inspect} missing from table. Dropping.")
+              attrs = attrs.except(*extra_attrs)
+            elsif (missing_attrs = (required_attributes - attr_names)).present?
```

Comment:
This `elsif` reads wrong. When `extra_attrs` are present, surely you still want to check independently that there are no `missing_attrs`?

Reply:
Oops.